docs/docs/script/caching.py

```diff
@@ -0,0 +1,343 @@
+
+
+# Again, nothing in the cache yet, but we the path is now in the "data" folder
+get_ipython().system('tree {ladi.fspath}')
+
+
+# Now let's look at just the first image from this dataset. Note that paths created by using the `ladi` root (e.g., by using the `/` operator below or calls like `iterdir` and `glob`) will inherit the same `Client` instance, and therefore the same `local_cache_dir` without our having to do extra work.
+
+flood_image = ladi / "DSC_0002_a89f1b79-786f-4dac-9dcc-609fb1a977b1.jpg"
+
+with flood_image.open("rb") as f:
+    i = Image.open(f)
+    plt.imshow(i)
+
+
+# Now just this one image file is in the cache
+get_ipython().system('tree {ladi.fspath}')
+
+
+# let's explicitly cleanup this directory, since it is not handled for us
+get_ipython().system('rm -rf data')
+
+
+# ## Accessing the cached version directly (read-only)
+# 
+# Many Python libraries don't properly handle `PathLike` objects. These libraries often only expect a `str` to be passed when working with files or, even worse, they will call `str(p)` on a Path that is passed before using it.
+# 
+# To use `cloudpathlib` with these libraries, you can pass `.fspath` which will provide the path to the cached version of the file as a string. 
+# 
+# **Warning:** Using the `.fspath` property will download the file from the cloud if it does not exist yet in the cache.
+# 
+# **Warning:** Since we are no longer in control of opening/closing the file, we cannot upload any changes when the file is closed. Therefore, you should treat any code where you use `fspath` as _read only_. Writes directly to `fspath` will not be uploaded to the cloud.
+# 
+
+# ## Handling conflicts
+# 
+# We try to be conservative in terms of not losing data—especially data stored on the cloud, which is likely to be the canonical version. Given this, we will raise exceptions in two scenarios:
+# 
+# `OverwriteNewerLocalError`
+# This exception is raised if we are asked to download a file, but our local version in the cache is newer. This likely means that the cached version has been updated, but not pushed to the cloud. To work around this you could remove the cache version explicitly if you _know_ you don't need that data. If you did write changes you need, make sure your code uses the `cloudpathlib` versions of the `open`, `write_text`, or `write_bytes` methods, which will upload your changes to the cloud automatically.
+# 
+# The `CloudPath.open` method supports a `force_overwrite_from_cloud` kwarg to force overwriting your local version.
+# 
+# `OverwriteNewerCloudError`
+# This exception is raised if we are asked to upload a file, but the one on the cloud is newer than our local version. This likely means that a separate process has updated the cloud version, and we don't want to overwrite and lose that new data in the cloud.
+# 
+# The `CloudPath.open` method supports a `force_overwrite_to_cloud` kwarg to force overwriting the cloud version.
+# 
+# 
+
+# # Clearing the file cache
```

Comment:
This and all following headers needed to be incremented a heading level. `h1` should only be used for the page title. This is also causing these sections not to show up in the TOC. 

![Screen Shot 2023-02-07 at 2 24 24 PM](https://user-images.githubusercontent.com/2721979/217344961-3bd73b1d-a988-4085-bebf-94088f9826a6.png)